lib/jekyll/tags/link.rb

```diff
@@ -11,21 +12,21 @@ def tag_name
 
       def initialize(tag_name, relative_path, tokens)
         super
-
         @relative_path = relative_path.strip
       end
 
       def render(context)
+        # additionally handle static files, that have a leading `/`
+        valid_paths = [@relative_path, "/#{@relative_path}"]
         site = context.registers[:site]
 
         site.each_site_file do |item|
-          return item.url if item.relative_path == @relative_path
-          # This takes care of the case for static files that have a leading /
-          return item.url if item.relative_path == "/#{@relative_path}"
+          rel_path = item.relative_path
+          return item.url if valid_paths.include?(rel_path)
```

Comment:
The ambiguity here worries me. This risks false positives.

Reply:
please elaborate.

Reply:
Sorry, I misread the code. Your code is correct...
...but more difficult to read. Why create the variable `rel_path`?

Reply:
Valid question. I had created `rel_path` in an earlier draft the line was
```ruby
return item.url if rel_path == @relative_path || rel_path == "/#{@relative_path}"
```
The above was flagged by rubocop which led to the current code..

I'll edit to make it simply
```ruby
return item.url if valid_paths.include?(item.relative_path)
```